src/views/mfa-verify/dropdown/FactorsDropDown.js

```diff
@@ -30,18 +30,8 @@ function (Okta, FactorsDropDownOptions, Factor) {
   return BaseDropDown.extend({
     className: 'bg-helper icon-button',
     screenReaderText: function () {
-      var factors = this.options.appState.get('factors'),
-          factor, factorLabel;
-      if (factors) {
-        factor = factors.findWhere(_.pick(this.options, 'provider', 'factorType'));
-        if (factor === undefined) {
-          //for factors that support cardinality and only have factorType
-          factor = factors.findWhere(_.pick(this.options, 'factorType'));
-        }
-      } else  {
-        factor = new Factor.Model(this.options.appState.get('factor'), this.toJSON());
-      }
-      factorLabel = factor.get('factorLabel');
+      var factor = FactorUtil.getCurrentFactor(_.extend(this.options, {toJSON: this.toJSON}));
```

Comment:
didn't realize in the beginning that we would need to pass the whole JSOn of this dropdown... not sure that is the right thing to do
What about
```
var factors = this.options.appState.get('factors'), factor;
if (factors) {
  factor = FactorUtil.findFactorInFactorsArray(factors, this.options.provider, this.options.factorType);
} else  {
  factor = new Factor.Model(this.options.appState.get('factor'), this.toJSON());
}
factorLabel = factor.get('factorLabel');
```

Reply:
we don't need everything.. But we do need appstate, provider and factorType besides toJSON. Should I limit to those options? I like how clear it is when we just use FactorUtil.getCurrentFactor

Reply:
Spoke to Mau offline. Changing this